packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -429,7 +432,15 @@ async function startWatcher(opts: SetupOpts) {
           pagesType: isAppPath ? PAGE_TYPES.APP : PAGE_TYPES.PAGES,
         })
 
-        if (isAppPath && isMetadataRoute(pageName)) {
+        if (
+          isAppPath &&
```

Comment:
maybe check `appDir` here so we could avoid the `!` in the filename replacement since it's the thing we care about